Werewolf for Telegram/Werewolf Node/Werewolf.cs

```diff
@@ -4626,63 +4631,140 @@ private bool DoGameEnd(ITeam team)
+                                        deathmessage += Environment.NewLine + GetLocaleString("ThiefEnd", thief.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString("TannerEnd", tann.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString($"{doppelganger.PlayerRole}End", doppelganger.GetName());
+                                    }
+                                }
+                                break;
+                            case 3:
+                                // Includes three of these roles: Tanner, Thief, Doppelganger, Sorcerer
+                                var doppelganger = alives.FirstOrDefault(x => x.PlayerRole == IRole.Doppelgänger);
+                                var thief = alives.FirstOrDefault(x => x.PlayerRole == IRole.Thief);
+                                var sorc = alives.FirstOrDefault(x => x.PlayerRole == IRole.Sorcerer);
+                                var tann = alives.FirstOrDefault(x => x.PlayerRole == IRole.Tanner);
+                                
+                                if (sorc != null)
+                                {
+                                    AddAchievement(sorc, AchievementsReworked.TimeToRetire);
+                                    deathmessage = GetLocaleString("SorcererEnd", sorc.GetName()) + Environment.NewLine;
+                                }
+                                
+                                if (tann != null)
+                                {
+                                    KillPlayer(tann, KillMthd.Suicide, killer: tann, isNight: false);
+                                    deathmessage += Environment.NewLine + GetLocaleString("TannerEnd", tann.GetName());
+                                }
+                                
+                                if (doppelganger != null)
+                                {
+                                    if (tann != null)
+                                    {
+                                        CheckDoppelganger();
+                                        if (doppelganger.PlayerRole == IRole.Tanner)
+                                        {
+                                            KillPlayer(doppelganger, KillMthd.Suicide, killer: doppelganger, isNight: false);
+                                        }
+                                    }
+                                    deathmessage += Environment.NewLine + GetLocaleString($"{doppelganger.PlayerRole}End", doppelganger.GetName());
+                                }
+                                
+                                if (thief != null)
+                                {
+                                    deathmessage += Environment.NewLine + GetLocaleString("ThiefEnd", thief.GetName());
+                                }
+                                break;
```

Comment:
In general, I think a lot of this copy-pasta can be avoided (since the game should not end anyway if there is a tanner left). So the changes should be a lot shorter.